src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -756,21 +773,23 @@ private static int compareImports(String import1, String import2) {
     }
 
     /**
-     * Counts empty lines before given.
-     * @param lineNo
+     * Counts empty lines between given.
+     * @param lineNoFrom
+     *        Line number of previous import.
+     * @param lineNoTo
      *        Line number of current import.
-     * @return count of empty lines before given.
+     * @return count of empty lines between given, exclusive, eg., (lineNoFrom, lineNoTo).
      */
-    private int getCountOfEmptyLinesBefore(int lineNo) {
+    private int getCountOfEmptyLinesBetween(int lineNoFrom, int lineNoTo) {
         int result = 0;
-        final String[] lines = getLines();
-        //  [lineNo - 2] is the number of the previous line
-        //  because the numbering starts from zero.
-        int lineBeforeIndex = lineNo - 2;
-        while (lineBeforeIndex >= 0
-                && CommonUtil.isBlank(lines[lineBeforeIndex])) {
-            lineBeforeIndex--;
-            result++;
+        if (lineNoFrom != lineNoTo) {
+            final String[] lines = Arrays.copyOfRange(getLines(), lineNoFrom, lineNoTo - 1);
```

Comment:
Documentation says exclusive, but are you including `lineNoFrom` by not doing `lineNoFrom + 1`?

Reply:
This isn't done.

Reply:
Documented the parameter `fromLineNo` and `toLineNo` as one-based:
```java
    /**
     * Counts empty lines between given parameters.
     * @param fromLineNo
     *        One-based line number of previous import.
     * @param toLineNo
     *        One-based line number of current import.
     * @return count of empty lines between given parameters, exclusive,
     *        eg., (fromLineNo, toLineNo).
     */
    private int getCountOfEmptyLinesBetween(int fromLineNo, int toLineNo) {
        ...
    }
``` 

As `Arrays#copyOfRange(T[] original, int from, int to)`, the parameter `from` is inclusive but the third parameter `to` is exclusive, then `Arrays.copyOfRange(getLines(), fromLineNo, toLineNo - 1);` is effectively coping range [fromLineNo - 1 + 1, toLineNo - 1), or [fromLineNo - 1 + 1, toLineNo - 1 - 1].

Comments added above `Arrays.copyOfRang(..)`.